File: backend/src/main/java/entities/Phase.java
package entities;

import java.time.LocalDate;

import javax.json.bind.annotation.JsonbTransient;
import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import io.quarkus.hibernate.orm.panache.PanacheEntity;

@Entity(name = "phase")
@Table(name = "phase")
public class Phase extends PanacheEntity {

    public Phase() {
    }

    public String name;
    public LocalDate start;
    public LocalDate end;
    public Double completed;

    @ManyToOne
    @JsonbTransient
    public Project project;

    @ManyToOne(optional = false)
    @JoinColumn(name = "project_id", nullable = false, updatable = false)
    public Project getProject() {
        return project;
    }

}

File: src/app/shared/model/Phase.ts
export class Phase {
  constructor(
    public name: string = '',
    public start: number = 0,
    public end: number = 0,
    public y: number = null,
    public completed: number = 0,
    public milestone: boolean = false
  ) {}
}

export interface PhaseDto {
  name: string;
  start: number;
  end: number;
  y: number;
  completed: number;
  milestone: boolean;
}
